### PR TITLE
perf: Make linting faster (no-changelog)

### PR DESCRIPTION
**Pull request by cstuncsik**

### CODE REVIEW

File: packages/@n8n_io/eslint-config/base.js [hunk 1]
```diff
@@ -322,6 +322,11 @@ const config = (module.exports = {
 		 */
 		'@typescript-eslint/triple-slash-reference': 'off', // @TECH_DEBT: Enable, disallowing in all cases - N8N-5820
 
+		/**
+		 * https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md
+		 */
+		'@typescript-eslint/naming-convention': 'off',
```
Comment (netroy) [inline packages/@n8n_io/eslint-config/base.js:328]: why did we disable this globally?

File: packages/cli/src/push/abstract.push.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { EventEmitter } from 'events';
 import { assert, jsonStringify } from 'n8n-workflow';
 import type { IPushDataType } from '@/Interfaces';
-import { Logger } from '@/Logger';
+import type { Logger } from '@/Logger';
```
Comment (netroy) [inline packages/cli/src/push/abstract.push.ts:4]: These types are used by decorator metadata, and this change will most likely break automatic dependency injection.

File: packages/@n8n_io/eslint-config/base.js (hunk 1 above)
Comment (cstuncsik) [reply packages/@n8n_io/eslint-config/base.js:328]: I don't why but after upgrading the lint-related packages eslint started to report `@typescript-eslint/naming-convention` as errors even though the eslint configs were not changed.

I'll give it another shot with removing it from here

File: packages/cli/src/push/abstract.push.ts (hunk 2 above)
Comment (cstuncsik) [reply packages/cli/src/push/abstract.push.ts:4]: This was another eslint reported error after upgrading lint-related packages and it was auto fixed but in this case it's better to turn off this rule for `cli` (and maybe for `core`) as well

**Review by netroy — APPROVED** [review]

**Pull request merged**